src/main/java/hudson/plugins/templateproject/UpdateHandler.java

```diff
@@ -0,0 +1,73 @@
+package hudson.plugins.templateproject;
+
+import hudson.model.AbstractProject;
+import hudson.scm.SCM;
+import hudson.util.DescribableList;
+
+import javax.annotation.Nonnull;
+import java.io.IOException;
+import java.util.logging.Level;
+import java.util.logging.Logger;
+
+
+/**
+ * @author Oleksandr Horobets
+ */
+public class UpdateHandler {
+    private static final Logger LOGGER = Logger.getLogger(UpdateHandler.class.getName());
+    private static UpdateHandler instance;
+
+    public void updateScm(@Nonnull AbstractProject project, @Nonnull String oldName, @Nonnull String newName) {
+        SCM scm = project.getScm();
+
+        if (scm instanceof ProxySCM) {
+            ProxySCM proxySCM = (ProxySCM) scm;
+
+            if (oldName.equals(proxySCM.getProjectName())) {
+                try {
+                    project.setScm(new ProxySCM(newName));
+                } catch (IOException e) {
+                    LOGGER.log(Level.SEVERE, "[TemplateProject] Unable to update SCM template reference", e);
+                }
+            }
+        }
```

Comment:
I've noticed that ProxySCM is a member of this plugin. BTW you're right that MultipleSCM is a potential case.